src/qt/button.cpp

```diff
@@ -46,7 +46,22 @@ bool wxButton::Create(wxWindow *parent, wxWindowID id,
 {     
     QtCreate(parent);
     SetLabel( wxIsStockID( id ) ? wxGetStockLabel( id ) : label );
-
+    
+    if( style & wxBU_LEFT )
+        m_qtPushButton->setStyleSheet( "Text-align:left" );
+    if( style & wxBU_RIGHT )
+        m_qtPushButton->setStyleSheet( "Text-align:right" );
+    if( style & wxBU_TOP )
+        m_qtPushButton->setStyleSheet( "Text-align:top" );
+    if( style & wxBU_BOTTOM )
+        m_qtPushButton->setStyleSheet( "Text-align:bottom" );
+    if( style & ( wxBU_LEFT | wxBU_RIGHT ) )
+        m_qtPushButton->setStyleSheet( "Text-align:center" );
+    if( style & ( wxBU_TOP | wxBU_BOTTOM ) )
+        m_qtPushButton->setStyleSheet( "Text-align:center" );
```

Comment:
This is completely wrong for a number of reasons:
- with this change, instead of supporting more cases, all of them are completely broken now. Do you get any alignment apart from the default (center) one? I'm guessing not;
- you're misunderstanding how styles can be combined: any combination of the values in `{left, right}` and `{top, bottom}` is valid;
- `center` alignment should never be set; it is the default, and simply not setting it will have the same effect; however setting it for an "unpermitted" value will make it different than the default style which is empty but does have the same effect.
- `else if`-s do have their benefic role; blindly removing all of them is at least undesirable;
- right now this is probably the only place where style sheet is set, but when another place will need one (i.e. maybe for setting image position vs text?), it will overwrite a previously set style sheet; the solution would be a way to construct the style sheet of all options and then set it only once.
- current `if( condition )` formatting is still different than both the default wxW formatting and the formatting already present in this file.